File: 06Adapter/Students/2018/kkorszun/TwoWayAdapter/src/main/scala/korszun/kacper/Seacraft.scala
package korszun.kacper

class Seacraft extends ISeacraft {
  private var speed = 0
  override def Speed: Int = speed

  override def IncreaseRevs: Unit = {
    speed += 10
    println("Seacraft engine increases revs to " + speed + " knots")
  }
}

File: 06Adapter/Students/2018/ZwaraPiotr/Adapter/Adapter.Zad1/Implementation/Seacraft.cs
using System;

using Adapter.Zad1.Abstraction;

namespace Adapter.Zad1.Implementation
{
    public class Seacraft : ISeacraft
    {
        public int Speed { get; private set; }

        public virtual void IncreaseRevs()
        {
            this.Speed += 10;
            Console.WriteLine("Seacraft engine increases revs to " + this.Speed + " knots");
        }
    }
}
